pkg/image/api/types.go

```diff
@@ -406,6 +409,40 @@ type ImageStreamImage struct {
 	Image Image
 }
 
+// ImageStreamTagCopy allows a client to create a copy of an existing image that changes
+// metadata or adds a new layer. It also allows the client to create a new image (i.e.
+// FROM scratch). The resulting image is stored as a tag on the stream.
+type ImageStreamTagCopy struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	metav1.ObjectMeta
+
+	// from is an optional reference to an existing image stream tag or image to copy.
+	// If from is not set, this is assumed to create a new scratch image.
+	From *kapi.ObjectReference
+	// image is metadata that will replace the existing metadata of from, or if from
+	// is empty, will be used to create a new scratch image.
+	Image *ImageCopyMetadata
```

Comment:
Not sure - this is more declarative than destination.  so if you specify it it's "this is what I want".